libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpAmrReader.java

```diff
@@ -0,0 +1,193 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package androidx.media3.exoplayer.rtsp.reader;
+
+import static androidx.media3.common.util.Assertions.checkArgument;
+import static androidx.media3.common.util.Assertions.checkNotNull;
+
+import androidx.media3.common.C;
+import androidx.media3.common.MimeTypes;
+import androidx.media3.common.util.Log;
+import androidx.media3.common.util.ParsableByteArray;
+import androidx.media3.common.util.Util;
+import androidx.media3.exoplayer.rtsp.RtpPacket;
+import androidx.media3.exoplayer.rtsp.RtpPayloadFormat;
+import androidx.media3.extractor.ExtractorOutput;
+import androidx.media3.extractor.TrackOutput;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+
+/**
+ * Parses an AMR byte stream carried on RTP packets and extracts individual samples. Interleaving
+ * mode is not supported.
+ */
+/* package */ final class RtpAmrReader implements RtpPayloadReader {
+  private static final String TAG = "RtpAmrReader";
+  /**
+   * The frame size in bytes, including header (1 byte), for each of the 16 frame types for AMR
+   * narrow band.
+   */
+  private static final int[] frameSizeBytesByTypeNb = {
+    13,
+    14,
+    16,
+    18,
+    20,
+    21,
+    27,
+    32,
+    6, // AMR SID
+    7, // GSM-EFR SID
+    6, // TDMA-EFR SID
+    6, // PDC-EFR SID
+    1, // Future use
+    1, // Future use
+    1, // Future use
+    1 // No data
+  };
+
+  /**
+   * The frame size in bytes, including header (1 byte), for each of the 16 frame types for AMR wide
+   * band.
+   */
+  private static final int[] frameSizeBytesByTypeWb = {
+    18,
+    24,
+    33,
+    37,
+    41,
+    47,
+    51,
+    59,
+    61,
+    6, // AMR-WB SID
+    1, // Future use
+    1, // Future use
+    1, // Future use
+    1, // Future use
+    1, // speech lost
+    1 // No data
+  };
+
+  private final RtpPayloadFormat payloadFormat;
+  private @MonotonicNonNull TrackOutput trackOutput;
+  private long firstReceivedTimestamp;
+  private int previousSequenceNumber;
+  private long startTimeOffsetUs;
+  private final int sampleRate;
+  private boolean isWideBand;
+
+  public RtpAmrReader(RtpPayloadFormat payloadFormat) {
+    this.payloadFormat = payloadFormat;
+    firstReceivedTimestamp = C.TIME_UNSET;
+    previousSequenceNumber = C.INDEX_UNSET;
+    this.sampleRate = this.payloadFormat.clockRate;
+    this.isWideBand = (payloadFormat.format.sampleMimeType == MimeTypes.AUDIO_AMR_WB);
```

Comment:
remove unnecessary parenthesis. I think here you also need to 
- do a checkNotNull() on sampleMimeType
- use this.payloadFormat here, or use payloadFormat on L96